modules/oe_content_featured_media_field/src/Plugin/Field/FieldWidget/FeaturedMediaAutocompleteWidget.php

```diff
@@ -0,0 +1,163 @@
+
+  /**
+   * Entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs FeaturedMediaAutocompleteWidget widget plugin.
+   *
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $field_definition
+   *   The definition of the field to which the widget is associated.
+   * @param array $settings
+   *   The widget settings.
+   * @param array $third_party_settings
+   *   Any third party settings.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   Entity type manager service.
+   */
+  public function __construct($plugin_id, $plugin_definition, FieldDefinitionInterface $field_definition, array $settings, array $third_party_settings, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($plugin_id, $plugin_definition, $field_definition, $settings, $third_party_settings);
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $plugin_id,
+      $plugin_definition,
+      $configuration['field_definition'],
+      $configuration['settings'],
+      $configuration['third_party_settings'],
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $element['featured_media'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Featured media'),
+    ];
+
+    $element['featured_media'] += parent::formElement($items, $delta, $element, $form, $form_state);
+    // Unset title_display to ensure the element title is always visible.
+    unset($element['featured_media']['target_id']['#title_display']);
+    $element['featured_media']['target_id']['#title'] = $this->t('Media item');
+    $element['featured_media']['target_id']['#description'] = $this->getMediaReferenceHelpText($element['featured_media']['target_id']['#selection_settings']);
+    $element['featured_media']['target_id']['#states'] = [
+      'required' => [
+        ':input[name="' . $this->fieldDefinition->getName() . '[' . $delta . '][featured_media][caption]"]' => ['filled' => TRUE],
+      ],
+    ];
+
+    $element['featured_media']['caption'] = [
+      '#title' => $this->t('Caption'),
+      '#description' => $this->t('The caption that goes with the referenced media.'),
+      '#type' => 'textarea',
+      '#default_value' => $items[$delta]->caption,
+      '#rows' => 2,
+      '#weight' => $delta,
```

Comment:
What's this weight doing here? You are ordering the caption and autocomplete between themselves input using the field delta?